### DISCUSSION TITLE
What is the time zone of the cron trigger if application runs in container in kubernetes cluster

### DISCUSSION THREAD

**Discussion by Eleonora-Vasylenko** [quartznet/quartznet#1491] [Q&A] [answered]
It is stated in the official docs, that the time zone, in which the cron trigger is fired, is the time zone of the system, on which the application runs. What is considered a "system time zone" if application runs in container in k8s cluster in Azure?
Say, if cron expression is specified as following 0 0 5 ? * * * (at 5am every day), will the trigger fire at 5am in the time zone of the cluster or there are other options?

**Comment by lahma** [comment, marked as answer]
It defaults to the local time zone if trigger configuration doesn't specify anything else:

https://github.com/quartznet/quartznet/blob/3d405c0c47b97995896d5aeecca114a62c67a45a/src/Quartz/Impl/Triggers/CronTriggerImpl.cs#L552-L565

Usually by default things run in UTC on Azure, at least the virtual machines. You could probably double-check by logging what's the value of TimeZoneInfo.Local on startup. So if you would like it to fire on 5AM on your time zone, you would need to configure it for the cron trigger.

**Reply by Eleonora-Vasylenko** [reply]
Thank you for the answer.
Yes, it is indeed UTC time zone, in which application in k8s cluster runs
